src/looper.ts

```diff
@@ -110,19 +110,23 @@ export function isIDValid(config: Config, ids: string): boolean {
   return true
 }
 
-export async function loopCheckSTUNServer(interval: number) {
-  let checkSTUNinterval
-  if (interval <= 0) {
-    // need to check 1 time only to get public IP
-    checkSTUNinterval = await getPublicIp()
-  } else {
-    checkSTUNinterval = setInterval(async () => {
-      await getPublicIp()
-    }, interval * MILLISECONDS)
+export async function loopCheckSTUNServer(interval: number, repeats: number) {
+  let counter = 0
 
-    if (process.env.CI || process.env.NODE_ENV === 'test') {
+  const checkSTUNinterval = setInterval(async () => {
+    if (counter === repeats) {
+      clearInterval(checkSTUNinterval)
+    } else if (interval <= 0) {
```

Comment:
CMIIW, say we have 1 probe with 5s interval, and the STUN check is 20s interval. So when `repeat` is 1, Monika will still wait for this to execute and finish even though the probe has finished?

Ideally, when `repeat` value exists, all other loop processes should be cancelled when the probes have been repeated x times.

Reply:
hmm so i have to check if there is a probe interval aborted, then abort the STUN loop also.

fixed, please recheck